app/modules/auth/auth.factory.js

```diff
@@ -6,18 +6,22 @@ require('./auth.module.js')
 var OAuth = require('panoptes-client/lib/oauth');
 
 // @ngInject
-function authFactory($interval, $timeout, $location, $window, localStorageService, ModalsFactory, zooAPI, zooAPIConfig, CribsheetFactory, $rootScope) {
+function authFactory($location, $rootScope, AnnotationsFactory, zooAPI, CribsheetFactory) {
 
     var factory;
 
     var _user = {};
 
     OAuth.checkCurrent()
-      .then(function (user) {
-        if (user) {
-            _setUserData();
-        }
-      });
+        .then(function (user) {
+            if (user) {
+                _setUserData();
+            }
+            return user;
+        })
+        .then(function(user) {
+            return zooAPI.beforeEveryRequest = _checkUserAndToken(user);
```

Comment:
This executes `_checkUserAndToken(user)` once then stores the return value in `zooAPI.beforeEveryRequest`.

I'm not sure why this function is returning the API client instance either.